src/main.rs

```diff
@@ -478,6 +479,403 @@ impl Config {
             target,
             toolchains_path,
             rustup_tmp_path,
+            client: Client::new(),
+        })
+    }
+
+    fn install(&self, bound: &Bound) -> anyhow::Result<()> {
```

Comment:
this commit is very noisy, maybe create impl blocks wherever there were functions before? It's ok to have multiple impl blocks

Reply:
thanks modified the commit to make it less noisy